### PR TITLE
Javascript Quiz Submission

### PR DESCRIPTION
**Pull request by caseywreed**
Go forth and grow

### CODE REVIEW

File: javascript/main.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
+var button = document.getElementById("button");
+button.addEventListener("click", makeTree);
+document.addEventListener("keydown", enter);
+
+
+function enter(e) {
+  if (13 == e.keyCode) {
+     makeTree()
```
Comment (sscotth) [inline javascript/main.js]: Make sure you pass the event so that the make tree function has access to it.

`makeTree(e)`

**Comment by sscotth** [comment]
Double check the file/folder structure you created. You duplicated the index.html and main.js files.

Id suggest a structure like this:

```
static-web-js-quiz-caseywreed
├── public or src or app
│   ├── index.html
│   └── main.js
└── README.md
```

 or this:

```
static-web-js-quiz-caseywreed
├── index.html
├── main.js
└── README.md
```

File: javascript/main.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  }
+}
+
+var tree = {}
```
Comment (sscotth) [inline javascript/main.js]: This line is not needed.

File: javascript/main.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+
+var button = document.getElementById("button");
+button.addEventListener("click", makeTree);
+document.addEventListener("keydown", enter);
+
+
+function enter(e) {
+  if (13 == e.keyCode) {
```
Comment (sscotth) [inline javascript/main.js]: While this works, this statement is considered a "yoda condition". I would avoid them.

https://en.wikipedia.org/wiki/Yoda_conditions

Try `if (e.keyCode == 13)`

File: javascript/main.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+function makeTree(tree) {
+  tree.height = document.getElementById("height").value;
+  tree.character = document.getElementById("character").value;
+	var counter = 1;
+
+	if (tree.character === "" || tree.height === "")
+		alert("ERROR: Both fields must have a value. You've ruined Christmas.")
+
+	for (i = 0; i < tree.height; i++) {
```
Comment (sscotth) [inline javascript/main.js]: always use `var` for your initial statement: `for (var i = 0; i < tree.height; i++) {`

File: javascript/javascript/main.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+
+var button = document.getElementById("button");
+button.addEventListener("click", makeTree);
+document.addEventListener("keydown", enter);
+
+
+function enter(e) {
```
Comment (sscotth) [inline javascript/javascript/main.js]: maybe calling this function `checkForEnter` would be more appropriate

File: javascript/main.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+
+var button = document.getElementById("button");
+button.addEventListener("click", makeTree);
+document.addEventListener("keydown", enter);
+
+
+function enter(e) {
```
Comment (sscotth) [inline javascript/main.js]: maybe calling this function `checkForEnter` would be more appropriate.

**Comment by sscotth** [comment]
Other than the minor things above, you have met the minimum requirements.

If you want to go further:
- Try using a form submit event and use only a single event listener.
- Try adding the counter variable to the for loop initialization, and final expressions. i.e.:

``` js
for (var i = 0, ????? ; i < tree.height ; i++, ?????) {
  console.log(" ".repeat(tree.height-i) + tree.character.repeat(counter))
}
```
- Try using a template string rather than string concatenation.

**Comment by gregkorte** [comment]
Good to merge. :+1:

**Pull request merged**